DDOT_module/child_classes/correction_data_generator.py

```diff
@@ -0,0 +1,397 @@
+
+                rev_map = anf.get_enumerated_rev_map_from_indices(cluster_i + cluster_j)
+
+                qubits_in_here_dict = dict(enumerate(qubits_in_here))
+
+                while qubits_in_here_dict != sorted_qubits_in_here:
+                    for index_qubit_hehe in range(len(qubits_in_here) - 1):
+                        if qubits_in_here[index_qubit_hehe] < qubits_in_here[index_qubit_hehe + 1]:
+                            pass
+                        elif qubits_in_here[index_qubit_hehe + 1] < qubits_in_here[
+                            index_qubit_hehe]:
+                            averaged_matrix = qrem_pt.permute_matrix(averaged_matrix,
+                                                                     len(whole_marginal),
+                                                                     [index_qubit_hehe + 1,
+                                                                      index_qubit_hehe + 2])
+
+                            anf.cool_print('Swapping qubits:', )
+                            print(qubits_in_here[index_qubit_hehe],
+                                  qubits_in_here[index_qubit_hehe + 1], index_qubit_hehe,
+                                  index_qubit_hehe + 1)
+
+                            qubits_in_here[index_qubit_hehe], qubits_in_here[
+                                index_qubit_hehe + 1] = qubits_in_here[index_qubit_hehe + 1], \
+                                                        qubits_in_here[index_qubit_hehe]
+
+                    qubits_in_here_dict = dict(enumerate(qubits_in_here))
+
+
+            else:
+                averaged_matrix = averaged_matrix_clusters_i_j
+
+            correction_matrix = np.linalg.inv(averaged_matrix)
+
+            string_marginal = ''.join(['q%s' % qqq for qqq in whole_marginal])
+            self._correction_indices[string_pair] = string_marginal
+            self._correction_indices['q%s' % i] = string_marginal
+            self._correction_indices['q%s' % j] = string_marginal
+            self._noise_matrices[string_marginal] = averaged_matrix
+            self._correction_matrices[string_marginal] = correction_matrix
+
+            if calculate_mitigation_errors:
+                # print(string_marginal, mitigation_errors_ij)
+                self._mitigation_errors[string_marginal] = mitigation_errors_ij
+
+                if mitigation_errors_ij / 2 >= 0.04:
+                    highly_correlated_qubits.append(
+                        {'qubits': string_marginal, 'error': mitigation_errors_ij / 2})
+
+    def get_pairs_correction_data(self,
+                                  pairs_list: list,
+                                  show_progress_bar: Optional[bool] = False):
+
+        # TODO FBM: possibly change resetting
+        self._noise_matrices = {}
+        self._correction_matrices = {}
+        self._mitigation_errors = {}
+        self._correction_indices = {}
+
+        if show_progress_bar:
```

Comment:
From my understanding there's just the same loop with and without tqdm. Have you considered following construction?
`
if show_progress_bar:
  from tqdm import tqdm
  pairs = tqdm(range(...))
else:
  pairs = range(...)

for pair in pairs:
  ...`

Reply:
This is great point, thanks.